### PR TITLE
parcel service migration script

### PR DESCRIPTION
**Pull request by zrshishir**
what about user_id and carrier_id? users table is not part of this parcel service.

### CODE REVIEW

File: cmd/migrations.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package cmd
+
+import (
+	"os"
+	postgres2 "parcel-service/internal/pkg/postgres"
+
+	_ "github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
```
Comment (shuza) [inline cmd/migrations.go:8]: what is the purpose of file at here?

**Review by shuza — COMMENTED** [review]
what is the purpose of side imports?

File: cmd/migrations.go (hunk 1 above)
Comment (zrshishir) [reply cmd/migrations.go:8]: This line `_"github.com/golang-migrate/migrate/v4/source/file"` is to import file driver in our project.

**Comment by zrshishir** [comment]
> what is the purpose of side imports?

@shuza  I am not clear about side imports.

**Comment by rdhawladar** [comment]
> > what is the purpose of side imports?
> 
> @shuza I am not clear about side imports.

@zrshishir 
This is side-effect import, _ means I need to import it but there is not direct use of it. Hope this link will help,
https://www.calhoun.io/why-we-import-sql-drivers-with-the-blank-identifier/

File: migrations/0001_add_tables.up.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+);
+
+CREATE TABLE IF NOT EXISTS carrier_request (
+    id serial PRIMARY KEY,
+    parcel_id int,
+    carrier_id int,
+    status int,
+    CONSTRAINT parcel_id
+        FOREIGN KEY(parcel_id)
+            REFERENCES parcel(id),
```
Comment (rdhawladar) [inline migrations/0001_add_tables.up.sql]: should use ON DELTE CASCADE. if parcel is deleted, there is no use of carrier_request.

Comment (zrshishir) [reply migrations/0001_add_tables.up.sql]: added delete cascade

**Review by rdhawladar — APPROVED** [review]
LGTM

File: migrations/0001_add_tables.up.sql [hunk 3]
```diff
@@ -0,0 +1,42 @@
+CREATE TABLE IF NOT EXISTS parcel_status (
+  id serial PRIMARY KEY,
+  status_value TEXT
+);
+
+CREATE TABLE IF NOT EXISTS parcel (
+  id serial PRIMARY KEY,
+  user_id int,
```
Comment (shuza) [inline migrations/0001_add_tables.up.sql]: use not null constant

File: migrations/0001_add_tables.up.sql [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  carrier_id int,
+  created_at timestamp,
+  updated_at timestamp,
+  status int,
```
Comment (shuza) [inline migrations/0001_add_tables.up.sql]: use not null constant

File: migrations/0001_add_tables.up.sql [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  source_address TEXT,
```
Comment (shuza) [inline migrations/0001_add_tables.up.sql]: use not null constant

File: migrations/0001_add_tables.up.sql [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  destination_address TEXT,
```
Comment (shuza) [inline migrations/0001_add_tables.up.sql]: use not null constant

File: migrations/0001_add_tables.up.sql [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  source_time timestamp,
+  type TEXT,
```
Comment (shuza) [inline migrations/0001_add_tables.up.sql]: use not null constant

File: migrations/0001_add_tables.up.sql [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+CREATE TABLE IF NOT EXISTS parcel (
+  id serial PRIMARY KEY,
+  user_id int,
+  carrier_id int,
+  created_at timestamp,
+  updated_at timestamp,
+  status int,
+  source_address TEXT,
+  destination_address TEXT,
+  source_time timestamp,
```
Comment (shuza) [inline migrations/0001_add_tables.up.sql:15]: what is the use of `source_time`?

**Review by shuza — CHANGES_REQUESTED** [review]
* Use not null constant in all required fields
* For `carrier_request` table primary key should be composite of `parcel_id` and `carrier_id`

File: migrations/0001_add_tables.up.sql (hunk 3 above)
Comment (zrshishir) [reply migrations/0001_add_tables.up.sql]: for user_id field?

Comment (zrshishir) [reply migrations/0001_add_tables.up.sql]: added not null constant for user_id and carrier_id

File: migrations/0001_add_tables.up.sql (hunk 8 above)
Comment (zrshishir) [reply migrations/0001_add_tables.up.sql:15]: source time is the time when a carrier will be assigned to carry the parcel.

**Review by Asad-noor — APPROVED** [review]

File: migrations/0001_add_tables.up.sql (hunk 8 above)
Comment (shuza) [reply migrations/0001_add_tables.up.sql:15]: is it required? what will be the benefit of this field?

**Review by shuza — APPROVED** [review]
lgtm

**Pull request merged**